src/homepageExperience/components/steps/cli/ExecuteAggregateQuery.tsx

```diff
@@ -27,6 +32,24 @@ export const ExecuteAggregateQuery = (props: OwnProps) => {
 
   const codeSnippet = `influx query 'from(bucket:"${bucketName}") |> range(start:-10m) |> mean()' --raw`
 
+  useEffect(() => {
+    document.addEventListener('keydown', fireKeyboardCopyEvent)
+    return () => document.removeEventListener('keydown', fireKeyboardCopyEvent)
+  }, [])
+
+  const userSelection = () => {
+    return window.getSelection().toString()
+  }
+
+  const fireKeyboardCopyEvent = event => {
+    if (
+      keyboardCopyTriggered(event) &&
+      userSelection().includes('influx query')
+    ) {
+      logCopyCodeSnippet()
+    }
+  }
+
```

Comment:
I think in all these components, this code should probably just be part of the `useEffect` function as well, since I think these are side effects